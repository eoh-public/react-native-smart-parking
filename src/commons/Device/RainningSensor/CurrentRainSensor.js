import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import { Colors } from '../../../configs';
import { CircleView } from '../../../commons';
import Text from '../../../commons/Text';
import IconRain from '../../../../assets/images/Device/rain.svg';
import IconSunny from '../../../../assets/images/Device/sunny.svg';

const CurrentRainSensor = memo(({ data }) => {
  const item = data.length
    ? data[0]
    : {
        evaluate: {},
      };
  const { text, raining, backgroundColor, borderColor } = item.evaluate;
  return (
    <View style={styles.standard}>
      <CircleView
        size={190}
        backgroundColor={backgroundColor}
        borderWidth={2}
        borderColor={borderColor}
        style={styles.center}
      >
        <LinearGradient
          style={styles.linearView}
          colors={[Colors.TransparentWhite, 'white']}
          start={{ x: 1, y: 0 }}
          end={{ x: 1, y: 1 }}
        />
        {raining ? <IconRain /> : <IconSunny />}
        {/* TODO: change this to general icon */}
        <Text style={styles.text}>{text}</Text>
      </CircleView>
    </View>
  );
});

export default CurrentRainSensor;

const styles = StyleSheet.create({
  standard: {
    flexDirection: 'row',
    marginTop: 16,
    justifyContent: 'center',
  },
  flatlistContent: {
    paddingHorizontal: 16,
  },
  center: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  linearView: {
    borderRadius: 95,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  text: {
    marginTop: 8,
    lineHeight: 32,
    fontSize: 24,
  },
});
