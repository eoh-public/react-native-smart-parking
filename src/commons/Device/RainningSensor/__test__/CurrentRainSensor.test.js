import React from 'react';
import renderer, { act } from 'react-test-renderer';
import CurrentRainSensor from '../CurrentRainSensor';

describe('Test CurrentRainSensor', () => {
  let tree;
  test('render CurrentRainSensor', () => {
    const data = [
      {
        evaluate: {
          text: 'Not raining',
          raining: false,
          backgroundColor: '#FEFFE6',
          borderColor: '#FADB14',
        },
      },
    ];
    act(() => {
      tree = renderer.create(<CurrentRainSensor data={data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render CurrentRainSensor raining: true', () => {
    const data = [
      {
        evaluate: {
          text: 'Not raining',
          raining: true,
          backgroundColor: '#FEFFE6',
          borderColor: '#FADB14',
        },
      },
    ];
    act(() => {
      tree = renderer.create(<CurrentRainSensor data={data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render CurrentRainSensor data empty', () => {
    act(() => {
      tree = renderer.create(<CurrentRainSensor data={[]} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
