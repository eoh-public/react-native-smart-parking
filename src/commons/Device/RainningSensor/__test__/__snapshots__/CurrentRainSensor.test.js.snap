// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CurrentRainSensor render CurrentRainSensor 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FEFFE6",
          "borderColor": "#FADB14",
          "borderRadius": 95,
          "borderWidth": 2,
          "height": 190,
          "width": 190,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          16777215,
          4294967295,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 95,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <SvgMock />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 24,
            "lineHeight": 32,
            "marginTop": 8,
          },
        ]
      }
    >
      Not raining
    </Text>
  </View>
</View>
`;

exports[`Test CurrentRainSensor render CurrentRainSensor data empty 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 95,
          "borderWidth": 2,
          "height": 190,
          "width": 190,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          16777215,
          4294967295,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 95,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <SvgMock />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 24,
            "lineHeight": 32,
            "marginTop": 8,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Test CurrentRainSensor render CurrentRainSensor raining: true 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FEFFE6",
          "borderColor": "#FADB14",
          "borderRadius": 95,
          "borderWidth": 2,
          "height": 190,
          "width": 190,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          16777215,
          4294967295,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 95,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <SvgMock />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 24,
            "lineHeight": 32,
            "marginTop": 8,
          },
        ]
      }
    >
      Not raining
    </Text>
  </View>
</View>
`;
